docs/installing-vyper.rst

```diff
@@ -178,75 +126,48 @@ If any unexpected errors or exceptions are encountered, please feel free to `ope
         export LDFLAGS="-L$(brew --prefix openssl)/lib"
         pip install scrypt
 
-    Now you can run the install and test commands again:
-    ::
-
-        make
-        make dev-deps
-        make test
+.. note::
 
     If you get the error ``ld: library not found for -lyaml`` in the output of `make`, make sure ``libyaml`` is installed using ``brew info libyaml``. If it is installed, add its location to the compile flags as well:
     ::
 
         export CFLAGS="-I$(brew --prefix openssl)/include -I$(brew --prefix libyaml)/include"
         export LDFLAGS="-L$(brew --prefix openssl)/lib -L$(brew --prefix libyaml)/lib"
+        
 
-    You can then run ``make`` and ``make test`` again.
-
-PIP
-***
-
-Each tagged version of vyper is also uploaded to `pypi <https://pypi.org/project/vyper/>`_, and can be installed using ``pip``.
-::
-
-    pip install vyper
-
-To install a specific version use:
-::
-
-    pip install vyper==0.1.0b2
-
-Docker
-******
-
-Dockerhub
-=========
+Windows
+--------
 
-Vyper can be downloaded as docker image from dockerhub:
-::
+Windows users can first `install Windows Subsystem for Linux <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`_ and then follow the instructions for Ubuntu, or `install Docker for Windows <https://docs.docker.com/docker-for-windows/install/>`_ and then follow the instructions for Docker.
 
-    docker pull vyperlang/vyper
+.. note::
+    - Windows Subsystem for Linux is only available for Windows 10.
+    - Windows versions that are < 10 and Windows 10 Home should install the slightly outdated `Docker Toolbox <https://docs.docker.com/toolbox/toolbox_install_windows/>`_, as explained in the link.
 
-To run the compiler use the `docker run` command:
-::
 
-    docker run -v $(pwd):/code vyperlang/vyper /code/<contract_file.vy>
+Creating a virtual environment
+==============================
 
-Alternatively you can log into the docker image and execute vyper on the prompt.
-::
+It is **strongly recommended** to install Vyper in **a virtual Python
+environment**, so that new packages installed and dependencies built are
+strictly contained in your Vyper project and will not alter or affect your
+other development environment set-up.
 
-    docker run -v $(pwd):/code/ -it --entrypoint /bin/bash vyperlang/vyper
-    root@d35252d1fb1b:/code# vyper <contract_file.vy>
 
-The normal paramaters are also supported, for example:
+To create a new virtual environment for Vyper run the following commands:
```

Comment:
Can we make the commands use the new `pyenv` approach as recommended above rather than manual creation of a virtualenv?

Reply:
poetry is really interesting, but I haven't used it yet. Is it very beginner friendly?

Reply:
@fubuloubu Very beginner friendly! Easy as Py. ;) I was using `pipenv` for a while but after many debates with a `poetry` evangelist peer I finally gave it a try and am now a convert. So, this is a vote for `poetry` for someone who went from manual `virtualenv` management to `mkvirtualenv` to `pipenv` to `poetry`. Give it a try and see what you think!